### PR TITLE
slow down passes drastically if towards our goal

### LINKED ISSUE #1190: If passing towards our goal, lower speed
We don't want our inability to catch a pass to cause us to score on our own goal. If the pass vector would shoot towards our goal, we want to greatly lower the power.

### PR DESCRIPTION
**Pull request by sem42198**
Closes #1190

### CODE REVIEW

File: soccer/gameplay/tactics/coordinated_pass.py [hunk 1]
```diff
@@ -155,6 +155,14 @@ def on_enter_preparing(self):
 
         kickpower = max(0.05, min(kickpower, 1.0))
 
+        pass_line = robocup.Line(main.ball().pos, self.receive_point)
+        our_goal = constants.Field.OurGoalSegment
+        intersect = our_goal.line_intersection(pass_line)
+        towards_goal = our_goal.dist_to(main.ball().pos) > our_goal.dist_to(self.receive_point) and not intersect is None
+        if towards_goal:
+            kickpower = max(0.05, kickpower * 0.25)
```
Comment (Jason27chan) [inline soccer/gameplay/tactics/coordinated_pass.py:163]: Would be good to test on real robots this sunday

**Comment by kylestach** [comment]
@tjones320 can we triage this? It looks good to me but I don't know how best to test it.

**Comment by kylestach** [comment]
@Jason27chan why was this closed? Was it supposed to be merged?

**Comment by Jason27chan** [comment]
Huh I thought it was merged.

**Comment by kylestach** [comment]
Okay, final bump on this. Is it likely to ever get merged? (Should it get merged in its current state?) If not I am going to close it.

@Jason27chan @tjones320 @HussainGynai

**Comment by kylestach** [comment]
Closed. Reopen later if we want this, but I don't think we've seen issues with it recently.

**Pull request closed without merging**